crates/fluvio-benchmark/src/benchmark_config/benchmark_settings.rs

```diff
@@ -0,0 +1,258 @@
+use std::{time::Duration, hash::Hash, fmt::Display};
+
+use fluvio::Compression;
+use serde::{Serialize, Deserialize};
+
+use super::benchmark_matrix::{RecordKeyAllocationStrategy, SharedSettings};
+use rand::{Rng, thread_rng, distributions::Uniform};
+
+#[derive(Debug, Serialize, Deserialize, Clone)]
+pub struct BenchmarkSettings {
+    pub topic_name: String,
+    /// Each sample is a collection of batches that all run on the same topic.
+    pub worker_timeout: Duration,
+    pub num_samples: usize,
+    pub duration_between_samples: Duration,
+    pub num_records_per_producer_worker_per_batch: u64,
+    pub producer_batch_size: u64,
```

Comment:
why duplicate fields?

Reply:
Yeah, the naming isn't good here. `producer_batch_size` is a fluvio configuration parameter, while `num_records_per_producer_worker_per_batch` is for the benchmarking batches.

Unflattening the config structure should help clarify, also I can try and think of better names.

Reply:
Moved to different parts of the config, more clarity in names.